### PR TITLE
Add docs for QueryApiKey API

### PR DESCRIPTION
**Pull request by ywangd**
This PR adds documentation for the new QueryApiKey API added in #75335 and #76144

doc-preview: https://elasticsearch_76521.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/security-api-query-api-key.html

Relates: #71023

### CODE REVIEW

File: x-pack/docs/en/rest-api/security/get-api-keys.asciidoc
Comment (ywangd) [inline x-pack/docs/en/rest-api/security/get-api-keys.asciidoc:194]: This change is not really related to the purpose of this PR. It's more of a bug fix of existing doc because this ID format is misleading.

File: x-pack/docs/en/rest-api/security/get-api-keys.asciidoc
Comment (ywangd) [inline x-pack/docs/en/rest-api/security/get-api-keys.asciidoc:182]: Same as above.

File: x-pack/docs/en/rest-api/security/query-api-key.asciidoc
Comment (tvernum) [inline x-pack/docs/en/rest-api/security/query-api-key.asciidoc:41]: Should we have some comments about the fields that can be queried? Maybe a list, or at least an explanation of how people would work out what the fields are.

File: x-pack/docs/en/rest-api/security/query-api-key.asciidoc
Comment (tvernum) [inline x-pack/docs/en/rest-api/security/query-api-key.asciidoc:50]:
```suggestion
(Optional, integer) The number of hits to return. Must not be negative.
```

File: x-pack/docs/en/rest-api/security/query-api-key.asciidoc
Comment (tvernum) [inline x-pack/docs/en/rest-api/security/query-api-key.asciidoc:54]: This is repeated from above. Is that intentional?

File: x-pack/docs/en/rest-api/security/query-api-key.asciidoc
Comment (tvernum) [inline x-pack/docs/en/rest-api/security/query-api-key.asciidoc:65]: I think we need to state the reverse as well.

> If the user has `manage_api_key` or higher (including `manage_security`), 
> this API returns API keys that are owned by any user

(any user? all users?)

File: x-pack/docs/en/rest-api/security/query-api-key.asciidoc
Comment (ywangd) [reply x-pack/docs/en/rest-api/security/query-api-key.asciidoc:54]: Yes it is indeed intentional. The same is repeated on the [Search API doc page](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html). The difference is that the two parameters, `from` and `size`, are further apart on the Search API page which makes the repitition more justifiable. I can consolidate them if you prefer.

File: x-pack/docs/en/rest-api/security/query-api-key.asciidoc
Comment (ywangd) [reply x-pack/docs/en/rest-api/security/query-api-key.asciidoc:65]: Good point. I added
> On the other hand, if the user
has `manage_api_key` or greater privilege (including `manage_security`), this
API returns API keys regardless of their owners.
?

File: x-pack/docs/en/rest-api/security/query-api-key.asciidoc
Comment (ywangd) [reply x-pack/docs/en/rest-api/security/query-api-key.asciidoc:41]: Thanks. I added a list to explain the available fields.

**Review by lockewritesdocs — COMMENTED** [review]
Suggested some changes -- will push some changes to this branch for a definition list under the `query` parameter.

File: x-pack/docs/en/rest-api/security/query-api-key.asciidoc
Comment (ywangd) [reply x-pack/docs/en/rest-api/security/query-api-key.asciidoc]: I moved it under the prerequisites as a separate bullet point.

**Comment by ywangd** [comment]
@elasticmachine update branch

**Review by lockewritesdocs — APPROVED** [review]
👍 with the latest changes included

**Comment by lockewritesdocs** [comment]
@elasticmachine update branch

**Comment by ywangd** [comment]
@tvernum Because I merged the rest spec PR (#76238) too soon, our docs build is currently broken because the rest spec file points to an URL yet to be added by this PR. To fix the issue, I am going to merge this PR once the CI passes. Both @lockewritesdocs and I are OK with where it is now. But I am happy to raise a follow-up PR if you spot any issues (commenting and review are still allowed by the UI after merging). Sorry for the inconvenience. I'll manage it better next time.

**Comment by ywangd** [comment]
@elasticmachine run elasticsearch-ci/part-2-fips

The CI failure is irrelevant but seems legit, I raised https://github.com/elastic/elasticsearch/issues/76689

**Pull request merged**
